scratch/PCA/extract_data.py

```diff
@@ -0,0 +1,123 @@
+
+# coding: utf-8
+
+# In[61]:
+
+#!/usr/bin/env python
```

Comment:
Do you intend for this to be a stand-alone executable?  If so, this will need to be the first line in the file.